crates/solidity/testing/snapshots/cst_output/MappingType/named_both/generated/0.4.11.yml

```diff
@@ -6,11 +6,18 @@ Source: >
 Errors: # 1 total
   - >
     Error: Expected EqualGreaterThan.
-       ╭─[crates/solidity/testing/snapshots/cst_output/MappingType/named_both/input.sol:1:16]
+       ╭─[crates/solidity/testing/snapshots/cst_output/MappingType/named_both/input.sol:1:15]
        │
      1 │ mapping(string Foo => bytes32 Bar)
```

Comment:
I noticed that when we create an error, we specify a range from current position till end of document. That can be quite large if the error is in the beginning of the file.

I realize we need to move to a DFA tokenizer for us to use the range of the current token. But until we get there, I wonder if it is better to limit it somehow, for example, simply clipping it to the end of the current line.

Otherwise, something like the HardHat LSP would paint the entire user window/file with red squiggles, making it much harder to read the code.

Reply:
I think we should do the simple (correct) thing, and leave that to the LSP author to maybe elide the erro range.

Reply:
Sounds good.